command/agent/config.go

```diff
@@ -229,6 +229,10 @@ type ClientConfig struct {
 	// AllocMountsDir is the directory for storing mounts into allocation data
 	AllocMountsDir string `hcl:"alloc_mounts_dir"`
 
+	// DHVPluginDir is the directory containing dynamic host volume plugins
+	// db TODO(1.10.0): document default directory is alongside alloc_mounts
+	DHVPluginDir string `hcl:"dhv_plugin_dir"` // db TODO(1.10.0): is this a good name?
```

Comment:
more bike-shedding;

1. where should it go by default? I put it next to `alloc_mounts` because the intention is for the plugins to put stuff in `alloc_mounts`, but it could go next to default `plugin_dir` within `data_dir` instead?
1. I put `plugins-` as a prefix, so it would sort next to other "plugin" dirs, but none are present by default where I've put this, and `host-volume-plugins` would better match the config value...
1. default `-`-joined `plugins-host-volume` or `_` `plugins_host_volume`?
1. changing CLI flag to `-host-volume-plugin-dir` seems good, eh?

so many little fiddly naming bits :sweat_smile:

Reply:
nice, I'll go with `<data-dir>/host_volume_plugins` then. it won't be sorted next to `plugins`, but I like the consistent word order in code and UX more than `ls` output of the data dir.  might even be good to type `/opt/nomad/data/hos<tab><tab>` so autocomplete won't fight with `plugins*`